### ISSUE TITLE
Cache: allow browsing / purging of content

### ISSUE THREAD

**Issue by clebergnu** [avocado-framework/avocado#4311]
The Avocado cache directories are organized by means of types:

* by_name
* by_location

And then within it, they're organized by a (obviously) one-way hash, examples:

```
by_location/
├── 06838d7fb3f29eff31af337f3077c4c5a3b3344b
│   ├── Fedora-Cloud-Base-32-1.6.aarch64_metadata.json
│   ├── Fedora-Cloud-Base-32-1.6.aarch64.qcow2
│   └── Fedora-Cloud-Base-32-1.6.aarch64.qcow2-CHECKSUM
├── 0f886b733da3fe2054b134a7dc49382b89a70ec3
│   ├── vmlinuz
│   └── vmlinuz-CHECKSUM
├── 152e9c9764dbfff13b555397f8697e3a1ab660c2
│   ├── u-boot-sunxi_2020.01%2Bdfsg-1_armhf.deb
│   └── u-boot-sunxi_2020.01%2Bdfsg-1_armhf.deb-CHECKSUM
├── 1ba70308b7a72db0d7580dd82dbe961fd33a636b
│   ├── Fedora-Cloud-Base-33-1.2.s390x_metadata.json
│   ├── Fedora-Cloud-Base-33-1.2.s390x.qcow2
│   └── Fedora-Cloud-Base-33-1.2.s390x.qcow2-CHECKSUM
├── 296e0b450673f95b3070ddf67181ad128417eb77
│   ├── kernel.img
│   └── kernel.img-CHECKSUM
```

We should allow users to view and purge the cache directories.  Example of use cases:

* Remove cached files not accessed during the last 30 days
* Remove cached files that are larger than 100 MiB
* Sort by last access, and remove all that exceed 4 GiB (that is, keep the last accessed 4 GiB worth of cached files)
* Is URL `http://foo.bar/baz.tar.gz` on cache?

While `vmimage` already stores some metadata for its own use, it should be integrated to a generic cache metadata repository.

---

QEMU: this would be used on a `make clean-artifacts` rule.

**Comment by beraldoleal** [comment]
One suggestion is to provide an avocado command line to handle those cases:

`avocado cache` or `avocado assets`

**Comment by willianrampazzo** [comment]
> One suggestion is to provide an avocado command line to handle those cases:
> 
> `avocado cache` or `avocado assets`

+1 

`avocado assets` plugin already has the structure to support more subcommands.

**Comment by clebergnu** [comment]
Most use cases have been implemented, and last one is tracked on #4457
